src/tokenizer.ts

```diff
@@ -3,98 +3,88 @@ import analyze from "./js.ts";
 export type TokenType = "string" | "tag" | "filter" | "comment";
 export type Token = [TokenType, string, number?];
 
-export interface TokenizeResult {
-  tokens: Token[];
-  position: number;
-  error: Error | undefined;
-}
```

Comment:
This is _technically_ a breaking change because of this signature change and removed export.

On that note, I've been feeling that too many internals are exposed - a change like this shouldn't have to be a breaking one. People should not be using this internal `tokenize` function or its signature. Generally I don't think users should have access to the tokenization process at all, but even if they wanted that, Vento already provides `env.tokenize()`. This many exports make things very hard to change, because people can be relying on effectively this entire codebase's structure. I think it would be good to, at some point in the future, remove all additional entrypoints and export a selection of user-facing tools from the main `mod.ts`.

What are your thoughts on this?

Reply:
The error remains, it's just thrown from the plugin itself instead of the tokenizer. And it should be decently easy to spot if the user has syntax highlighting (which I imagine almost all users do).

+1 to holding the documentation as "source of truth", but it remains a bit of a pain point from a maintenance perspective since users do rely on importing from specific locations right now as opposed to importing everything from `mod.ts`. If we want to move things around, we'll be breaking people's environments or have unnecessary duplicate exports scattered throughout the codebase. I don't think we can change this safely now, but something to keep in mind for a next major version.

Reply:
Okay.